libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -1116,11 +1116,40 @@ AP_Logger::log_write_fmt *AP_Logger::msg_fmt_for_name(const char *name, const ch
         return nullptr;
     }
     f->msg_type = msg_type;
-    f->name = name;
-    f->fmt = fmt;
-    f->labels = labels;
-    f->units = units;
-    f->mults = mults;
+
+    if (copy_strings) {
+        // cannot use pointers to memory that might move, must allocate and copy
+        struct log_write_fmt_strings *ls_copy = (struct log_write_fmt_strings*)malloc(sizeof(log_write_fmt_strings));
+        if (ls_copy == nullptr) {
+            free(f);
+            return nullptr;
+        }
+
+        strcpy(ls_copy->name, name);
+        strcpy(ls_copy->format, fmt);
+        strcpy(ls_copy->labels, labels);
```

Comment:
```suggestion
        strncpy(ls_copy->name, name, sizeof(ls_copy->name));
        strncpy(ls_copy->format, fmt, sizeof(ls_copy->format));
        strncpy(ls_copy->labels, labels, sizeof(ls-copy->labels));
```

... and elsewhere

Reply:
```
../../libraries/AP_Logger/AP_Logger.cpp:1128:16: error: 'char* strncpy(char*, const char*, size_t)' specified bound 5 equals destination size [-Werror=stringop-truncation]
 1128 |         strncpy(ls_copy->name, name, sizeof(ls_copy->name));
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
@peterbarker We could either back to `strcpy` or disable the error for this section. Or maybe there is a third solution? What do you think?

Reply:
`strncpy_noterm`

Reply:
perfect, thanks!